archiver/archiver.py

```diff
@@ -639,26 +646,28 @@ def send_accessions(self, accessions: List[IngestAccession]):
             for accession in accessions:
                 entity_type, entity_id = self.ingest_api.entity_info_from_url(accession.ingest_url)
                 ingest_entity = self.ingest_api.get_entity_by_id(entity_type, entity_id)
-                entity_patch = IngestArchiver.generate_patch(accession.ingest_type, ingest_entity, accession.accession_id)
+                entity_patch = IngestArchiver.generate_patch(accession, ingest_entity)
                 try:
                     self.ingest_api.patch_entity_by_id(entity_type, entity_id, entity_patch)
                 except HTTPError:
                     logging.error("Failed to send to ingest", HTTPError)
 
     @staticmethod
-    def generate_patch(ingest_type, ingest_entity, accession_id):
+    def generate_patch(accession: IngestAccession, ingest_entity):
         entity_patch = {'content': ingest_entity['content']}
-        # todo: what about studies?
-        if ingest_type == 'project':
-            entity_patch['content']['biostudies_accessions'] = accession_id
-            entity_patch['content']['insdc_project_accessions'] = accession_id
-            # patch['content']['insdc_study_accessions'] = accession.accession
-        elif ingest_type == 'biomaterial':
-            entity_patch['content']['biomaterial_core']['biosamples_accession'] = accession_id
-        elif ingest_type == 'process':
-            entity_patch['content']['insdc_experiment'] = accession_id
-        elif ingest_type == 'file':
-            entity_patch['content']['insdc_run_accessions'] = accession_id
+        if accession.accession_type == 'project':
+            entity_patch['content']['biostudies_accessions'] = accession.accession_id
+        elif accession.accession_type == 'study':
+            entity_patch['content']['insdc_project_accessions'] = accession.accession_id
+            # DSP returns study_accessions, but an error in HCA metadata requires we store them as project_accessions
+            # Once this error is fixed we should also retrieve the project accession from ENA using the study accession
+            # entity_patch['content']['insdc_study_accessions'] = accession.accession_id
+        elif accession.accession_type == 'biomaterial':
+            entity_patch['content']['biomaterial_core']['biosamples_accession'] = accession.accession_id
+        elif accession.accession_type == 'process':
+            entity_patch['content']['insdc_experiment'] = accession.accession_id
+        elif accession.accession_type == 'file':
+            entity_patch['content']['insdc_run_accessions'] = accession.accession_id
```

Comment:
the accession path in the json can be configurable and have a method that will inject the value in a json based on the path.